wapitiCore/net/jsparser_angular.py

```diff
@@ -0,0 +1,81 @@
+import enum
+import re
+from urllib.parse import urlparse
+
+class JsParserAngular:
+    """
+    A class that try to parse JS files from angular webapps
+    and extract interesting subdomains and paths.
+    """
+
+    def __init__(self, url: str, data: str):
+        """
+        Create a new JsParserAngular object.
+
+        Takes the following arguments:
+            url: The absolute URL of the JS file on the server
+            data: The content of the JS file as a string
+        """
+        self.links = []
+
+        parts = urlparse(url)
+        netloc = parts.netloc
+        self.target_domain = netloc
+
+        if netloc and len(netloc.split('.')) > 2:
```

Comment:
You should not extract domain by splitting on dots, it won't work for .co.uk suffixes. Instead use `tld.get_fld` which is already used elsewhere in Wapiti.

Also you should check that parts.hostname is not an IP address.